knative-operator/pkg/common/monitoring.go

```diff
@@ -36,6 +42,36 @@ func SetupMonitoringRequirements(api client.Client, instance mf.Owner) error {
 	return nil
 }
 
+func RemoveOldServiceMonitorResources(namespace string, api client.Client) error {
+	currentSMName := os.Getenv(operatorServiceMonitorNameEnvKey)
+	if currentSMName == "" {
+		currentSMName = operatorServiceMonitorDefaultName
+	}
+	smList := monitoringv1.ServiceMonitorList{}
+	if err := api.List(context.TODO(), &smList, client.InNamespace(namespace)); err != nil {
+		return err
+	}
+	for _, sm := range smList.Items {
+		// Skip the sm that is being installed with the current operator version
+		if sm.Name != currentSMName && strings.HasPrefix(sm.Name, "knative-openshift-metrics-") {
+			// Delete service monitor and the related service monitor service, skip error if not found
+			if err := api.Delete(context.TODO(), sm); err != nil && !errors.IsNotFound(err) {
```

Comment:
```suggestion
			if err := api.Delete(context.Background(), sm); err != nil && !errors.IsNotFound(err) {
```

Reply:
I was not sure if I could pass any ctx later on thus TODO but Background probably makes more sense.

Reply:
Yeah the line is somewhat blurry, but TODO shouldn't be in any production code (IMO at least). It's supposed to trigger linter warnings I guess :shrug: